Firestore/core/test/unit/remote/serializer_test.cc

```diff
@@ -1489,7 +1489,7 @@ TEST_F(SerializerTest, EncodesResumeTokens) {
   core::Query q = Query("docs");
   TargetData model(q.ToTarget(), 1, 0, QueryPurpose::Listen,
                    SnapshotVersion::None(), SnapshotVersion::None(),
-                   Bytes({1, 2, 3}));
+                   Bytes({1, 2, 3}), absl::nullopt);
```

Comment:
Please add (at least) one test case here to test changes that you made to `serializer.cc`. In fact, each of the code changes should be accompanied by a unit test. Please look into adding unit test coverage for all of the changes that you've made in this PR. If one of them seems "too hard" then let me know and we'll discuss.